coreneuron/apps/main1.cpp

```diff
@@ -402,6 +416,20 @@ std::unique_ptr<ReportHandler> create_report_handler(ReportConfiguration& config
     return report_handler;
 }
 
+/**
+ * \brief Installs a SIGTERM handler so that we finish the current simulation without losing data
+ * \return True if a checkpoint was performed. False otherwise (not enough elapsed time)
+ */
+static void install_sigterm_handler() {
+    auto sigh = [](int) {
+        std::cerr << "SIGTERM caught! Halting sim and dumping checkpoint" << std::endl;
+        coreneuron::stoprun = true;
+    };
```

Comment:
What is often do is to enable a "double sigterm", to really cut things

How long is this checkpoint can be?